File: code/ex1029.java
// Sort these people by the money the company can save,
// if ask that person to go to city A instead of city B

import java.util.*;

class Solution {
    public int twoCitySchedCost(int[][] costs) {
        Arrays.sort(costs, new Comparator<int[]>() {
            @Override
            public int compare(int[] a, int[] b) {
                return (b[1] - b[0]) - (a[1] - a[0]) ;
            }
        });
        
        int res = 0;
        for (int i = 0; i < costs.length / 2; i++) {
            res = res + costs[i][0] + costs[costs.length - 1 - i][1];
        }
        
        return res;
    }
}

File: code/ex1029.cpp
struct cmp {
    bool operator()(vector<int>& c1, vector<int>& c2) {
        return abs(c1[0] - c1[1]) > abs(c2[0] - c2[1]);
    }
};

class Solution {
public:
    int twoCitySchedCost(vector<vector<int>>& costs) {
        int n = costs.size();
        sort(costs.begin(), costs.end(), cmp());
        
        int A = 0, B = 0, cost = 0;
        for (int i = 0; i < n; i++) {
            if (A >= n / 2)    cost += costs[i][1];
            else if (B >= n / 2)    cost += costs[i][0];
            else {
                if (costs[i][0] > costs[i][1])    {B++;    cost += costs[i][1];}
                else    {A++;    cost += costs[i][0];}
            }
        }
        
        return cost;
    }
};